### ISSUE TITLE
Support E0435 error detection

### ISSUE THREAD

**Issue by weihe0** [intellij-rust/intellij-rust#5427]
## Environment

* **IntelliJ Rust plugin version:** 0.2.122.3105-201
* **Rust toolchain version:** 1.43.1 (8d69840ab 2020-05-04) x86_64-pc-windows-msvc
* **IDE name and version:** IntelliJ IDEA 2020.1.1 Ultimate Edition (IU-201.7223.91)
* **Operating system:** Windows 10 10.0

## Problem description

Defining an array with a non-constant length, such as let a: [i32; n] = [0; n];,  is prohibited in Rust, but Intellij Rust does not detect the error. See [E0435](https://doc.rust-lang.org/error-index.html#E0435)
 
## Steps to reproduce
Write a simple program
```rust
fn main() {
    let n = 5;
    let a: [i32; n] = [0; n];
    println!("{}", a[0]);
}
```
IntelliJ Rust does not detect the array initialization error, so it thinks it is a valid program. However, when trying to run it, rustc says that, in let a: [i32; n], n is not allowed because it is a non-constant value.
